config/v1/types_infrastructure.go

```diff
@@ -1082,9 +1082,41 @@ type VSpherePlatformStatus struct {
 	LoadBalancer *VSpherePlatformLoadBalancer `json:"loadBalancer,omitempty"`
 }
 
+// IBMCloudServiceEndpoint stores the configuration of a custom url to
+// override existing defaults of IBM Cloud Services.
+type IBMCloudServiceEndpoint struct {
+	// Name is the name of the IBM Cloud service.
+	// For example, the IBM Cloud Private IAM service could be configured with the
+	// service `Name` of `Iam` and `URL` of `https://private.iam.cloud.ibm.com`
+	// Whereas the IBM Cloud Private VPC service for US South (Dallas) could be configured
+	// with the service `Name` of `Vpc` and `URL` of `https://us.south.private.iaas.cloud.ibm.com`
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^[a-z0-9-]+$`
+	// +kubebuilder:validation:MaxLength=32
+	Name string `json:"name"`
+
+	// URL is fully qualified URI with scheme https, that overrides the default generated
+	// endpoint for a client.
+	// This must be provided and cannot be empty.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=uri
+	// +kubebuilder:validation:Pattern=`^https://`
+	URL string `json:"url"`
+}
+
 // IBMCloudPlatformSpec holds the desired state of the IBMCloud infrastructure provider.
 // This only includes fields that can be modified in the cluster.
-type IBMCloudPlatformSpec struct{}
+type IBMCloudPlatformSpec struct{
+	// ServiceEndpoints is a list of custom endpoints which will override the default
+	// service endpoints of an IBM Cloud service.
```

Comment:
Is there a mechanism to copy the accepted service endpoints from the spec to status?

Reply:
Currently no. Should I drop the Spec until we implement the CCM to process and validate the Spec before updating the Status at runtime?

Reply:
I think they need to go into the same release, so if you think you can get the API and validation into the same release, it's ok to keep them, if not, may be easier to start with just status

Reply:
I'll start with just Status then, and drop Spec. Since the freeze is this week, and I wouldn't expect to get the CCM changes in before that.